### DISCUSSION TITLE
Variable wird nicht aktualisiert

### DISCUSSION THREAD

**Discussion by Pscharumbel** [SukramJ/aiohomematic#1883] [Q&A]
Hi,
ich weiß nicht, ob es ein Fehler ist oder eine Einstellungssache, daher habe ich vorsichtshalber kein issue eröffnet.
Seit der Version 1.71 werden die Systemvariablen nicht mehr richtig aktualisiert. Oder zumindest nicht alle. Ich habe eine Variable namens „Fenster“ mit drei String-Werten. Wird die Variable in der CCU von Wert 1 auf 2 und später von 2 auf 3 gesetzt, wird das in HA korrekt angezeigt. Ändert sich die Variable in der CCU von 3 auf 1, bleibt sie in HA aber auf 3. Erst ein Neustart der Homematic-Integration behebt den Fehler.
Es ist dabei egal, ob die Variable den hahm-Text enthält (habe die Variable dazu testweise neu angelegt).
Wenn ich andersrum den Wert in HA ändere, wird es in der CCU korrekt übernommen.
Bin wieder zurück auf 1.70 da läuft alles wie es soll.
Eine Idee woran es liegen könnte?
Viele Grüße,
Jens

**Comment by SukramJ** [comment]
Alle Systemvariablen werden **in einem Abruf** alle **30 Sekunden** geladen. Nur wenn sich der Wert **ändert** wird auch die Systemvariable ge**ändert**.

**Reply by Pscharumbel** [reply]
Hi! 
Danke für die Antwort. Der Wert der Variablen ändert sich aber auch nach 30 Minuten nicht, obwohl er sich in der CCU geändert hat.

**Reply by SukramJ** [reply]
Ich kann das auf meinem System nicht nachvollziehen.
Ändern sich andere Systemvariablen? 
Gibt es Meldungen im HA Protokoll?

**Reply by Pscharumbel** [reply]
Andere Variablen funktionieren.
Hatte die problematische Variable aber testweise neu angelegt, wo das Problem auch auftrat. Name „Fenster“ bzw. „Fenster2“ mit den Werten „alle geschlossen“, „mind. 1 Fenster offen“, „Winterwarnung“.
Im Protokoll kann ich diesbezüglich nichts finden.

**Reply by SukramJ** [reply]
Ist die neue Variable ggf. noch deaktiviert?

**Reply by Pscharumbel** [reply]
Nein

**Reply by SukramJ** [reply]
Mach mal einen Screen shot von der Systemvariable in der CCU, bzw. wie Du diese anlegst.

**Reply by Pscharumbel** [reply]
![IMG_0597](https://github.com/user-attachments/assets/396a4a9a-3f67-4d49-b487-2f760c57352b)

**Reply by Pscharumbel** [reply]
Lief ja alles seit fast 2 Jahren problemlos mit der Integration. Erst seit 1.71 nicht.

**Reply by SukramJ** [reply]
> Lief ja alles seit fast 2 Jahren problemlos mit der Integration. Erst seit 1.71 nicht.

Interessiert mich eigentlich nicht!
Auch mit der Version 1.71 hat sich die Verarbeitung der Systemvariablen nicht geändert.

**Reply by SukramJ** [reply]
Aktuell ist übrigens Version 1.72.0

**Reply by Pscharumbel** [reply]
1.72 hab auch probiert. Auch das 1.73 prerelease. Überall das gleiche.

**Reply by SukramJ** [reply]
Ich kann ein Problem nur bei `hahm` Systemvariablen feststellen, wenn nach dem Neustart/Reload der Wert mind. einmalig in HA geändert wurde.

**Reply by Pscharumbel** [reply]
Danke für deine Mühe!
Hatte das Problem auch ohne hahm. Hatte es dann erst hinzugefügt, in der Hoffnung, es damit zu umgehen. Änderte aber nichts.

**Reply by SukramJ** [reply]
Mit der neuen 1.73.0b2 hab ich weder mit noch ohne `hahm` Probleme mit der Aktualisierung bei Wertelisten Sysvars mit den Werten alle `geschlossen;mind. 1 Fenster offen;Winterwarnung`.

**Reply by Pscharumbel** [reply]
Funktioniert bei mir auch wieder bestens. 
Vielen Dank!
